### ISSUE TITLE
GCC compile note in editor/editor_settings.cpp: -Wmisleading-indentation is disabled from this point onwards, since column-tracking was disabled due to the size of the code/headers

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#26443]
**Godot version:**
2363c38

**OS/device including version:**
Mageia 7 x86_64, GCC 8.3.0

**Issue description:**
When compiling the current master branch with GCC 8, this note shows up:
```
editor/editor_settings.cpp: In member function 'void EditorSettings::_get_property_list(List<PropertyInfo>*) const':
editor/editor_settings.cpp:188: note: -Wmisleading-indentation is disabled from this point onwards, since column-tracking was disabled due to the size of the code/headers
   _EVCSort vc;
```

It's harmless and doesn't count as a warning (so doesn't fail on `-Werror`), but it's still a slight annoyance, especially if it means that actual warnings about `-Wmisleading-indentation` will be missed.

I looked at the incriminated file and I don't see anything wrong with the indentation, so I'm not sure how to solve it. Any idea @marxin?

**Steps to reproduce:**
- Build with GCC 8, `scons p=x11`

**Comment by marxin** [comment]
> I looked at the incriminated file and I don't see anything wrong with the indentation, so I'm not sure how to solve it. Any idea @marxin?

I'm on it. Apparently GCC looses a column of a expression. For a pre-processed source file I see:
```
mi.ii:23809: note: -Wmisleading-indentation is disabled from this point onwards, since column-tracking was disabled due to the size of the code/headers
```

We should not give up after 20K of lines. I'm reducing that and will report to GCC soon.

**Comment by marxin** [comment]
@akien-mga Please assign it to me.

**Comment by marxin** [comment]
Upstream bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89549
It's related to fact that the line is very wide (comes from a macro), which can be changed..

**Comment by marxin** [comment]
It's cause by huge body of ```#define GDCLASS(m_class, m_inherits)```. Which is then expanded on a single line. Anyway, GCC should handle that..

**Comment by RevoluPowered** [comment]
Getting this on GCC too

**Comment by marxin** [comment]
> Getting this on GCC too

Unfortunately, we haven't made any progress about this bug in the GCC compiler.

**Comment by akien-mga** [comment]
Fixed by #40332.
